data/trie/leafNode.go

```diff
@@ -241,3 +241,26 @@ func (ln *leafNode) print(writer io.Writer, index int) {
 
 	_, _ = fmt.Fprintf(writer, "L:(%s - %s)\n", key, val)
 }
+
+func (ln *leafNode) deepClone() node {
```

Comment:
You should check if ln.isEmptyOrNil(), as all the other methods do.

Reply:
Checked only for nil